pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -409,26 +410,29 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(
 				Name:     ingressListener.Port.Name,
 			}
 
-			// if app doesn't have a declared ServicePort, but a sidecar ingress is defined - we can't generate a listener
-			// for that port since we don't know what policies or configs apply to it ( many are based on service matching).
-			// Sidecar doesn't include all the info needed to configure a port.
+			bind := ingressListener.Bind
+			if len(bind) == 0 {
```

Comment:
This affect both instance == nil and instance != nil. Could you limit the new getSidecarInboundBindIP to instance != nil? 

Although I have no idea when it would fallback to 127.0.0.1 or 0.0.0.0

Reply:
See previous response. We should not limit. Its also in the comments around this code. Whether or not we find a service instance for the sidecar.ingressListener port, we should always try to bind to one of the IPs of the proxy (if the user did not specify any bind IP). If we can't locate a suitable proxy IP to bind to, we should fallback to loopback socket, but not 0.0.0.0

Reply:
why should we limit it to when instance != nil ? I could declare this on a VM and request that it be bound to some wildcard IP..